plugin/SearchBarItem.qml

```diff
@@ -0,0 +1,243 @@
+
+        property bool iconized: true
+        property int modelCount: -1
+
+        property real _prevNextWidth: modelCount > 0 ? (Theme.itemSizeSmall + Theme.paddingMedium)*2 : 0.
+        implicitWidth: !iconized ? _editor.implicitWidth + Theme.paddingSmall
+                                   + Theme.itemSizeSmall*2  // width of two icons
+                                   + _prevNextWidth // width of prev / next icons
+                                 : Theme.itemSizeSmall
+        height: Math.max(Theme.itemSizeMedium, _editor.height + Theme.paddingMedium + Theme.paddingSmall)
+
+        anchors.verticalCenter: parent.verticalCenter
+        anchors.left: parent.left
+        anchors.right: parent.right
+        property real _margin: Math.max((root.iconizedWidth - Theme.itemSizeSmall) / 2., 0.)
+        anchors.leftMargin: _margin
+        anchors.rightMargin: _margin
+
+        focusOutBehavior: FocusBehavior.ClearPageFocus
+        font {
+            pixelSize: Theme.fontSizeLarge
+            family: Theme.fontFamilyHeading
+        }
+
+        textLeftMargin: Theme.itemSizeSmall + Theme.paddingMedium + Theme.horizontalPageMargin - Theme.paddingLarge
+        textRightMargin: Theme.itemSizeSmall + Theme.paddingMedium + Theme.horizontalPageMargin - Theme.paddingLarge + _prevNextWidth
+        textTopMargin: labelVisible ? Theme.paddingSmall : (height/2 - _editor.implicitHeight/2)
+        labelVisible: modelCount > 0 && !searchField._editor.activeFocus
+        Binding on label {
+            target: searchField
+            //% "%n item(s) found"
+            value: qsTrId("sailfish-office-lb-%n-matches", modelCount)
+            when: modelCount > 0
+        }
+
+        onModelCountChanged: if (modelCount == 0) text = "" // Allow the placeholder
+        // to be visible for no results.
+        onIconizedChanged: if (!iconized) _editor.forceActiveFocus()
+
+        //: Placeholder text of SearchField
+        placeholderText: (modelCount == 0)
+            //% "No result"
+            ? qsTrId("sailfish-office-search-no-result")
+            //% "Search on document"
+            : qsTrId("sailfish-office-search-document")
+        onFocusChanged: {
+            if (focus) {
+                cursorPosition = text.length
+                searchField.iconized = false
+            }
+        }
+
+        inputMethodHints: Qt.ImhNoAutoUppercase | Qt.ImhPreferLowercase | Qt.ImhNoPredictiveText
+        EnterKey.iconSource: text != "" ? "image://theme/icon-m-enter-accept"
+                                        : "image://theme/icon-m-enter-close"
+        EnterKey.onClicked: {
+            focus = false
+            if (text != "") {
+                root.requestSearch(text)
+            } else {
+                iconized = true
+            }
+        }
+
+        background: null
+
+        Item {
+            parent: searchField // avoid TextBase contentItem auto-parenting
+            anchors.fill: parent
```

Comment:
Yes, at first, the idea was to stay as close to the source code of the SearchField item as possible to allow a hypothetical merge in future. The main variations are the introduction of the prev / next buttons and the ability to catch the press callback on the clear button and adapt its behaviour. All the rest (more or less) can be moved outside such a modified SearchField (the use of the label, the iconized property…).

So here, depends where we want to go:
- either we keep the code as close as possible to the SearchField item, which means putting the icons in the TextField and play with text margins.
- or we implement a SearchBarItem made from a TextField and separated IconButtons, which would allow to be able to play easily with the visibility of each elements.